### ISSUE TITLE
Apply Audio In in other spots

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#1795]
Like: Wouldn't it be cool if a downsampled audio input was available as an LFO modulation source?  Korridor had a bunch of ideas he can attach here.

**Comment by mkruselj** [comment]
u-he Zebrify has a terrific implementation of this. Basically the input goes through envelope detector, and pitch detector, and then you get outputs of those as modulation sources: Envelope Follower, Pitchiness, Essness, Transient.

**Comment by K0rrid0r** [comment]
@mkruselj wow yeah! that is exactly how i was thinking this could be implemented but your explanation here was excellent. I'm going to look into how that can be applied in Zebrify and see if it will make sense to have in surge at well placed points, because it should make sense somehow.

Actually any slider could use this trigger functionality and also anything that can be pressed. At first maybe a few well choosed points will be fine.

**Comment by baconpaul** [comment]
Right. Getting an oscillator as a modulation source (with appropriate envelopes) is the most “surge”-y way of doing it indeed. Huh neat. 

I can imagine a world where OSC3 has a special mode triggered in the mixer where the audio is also fed into the 4 audio modulation sources (as you say, direct averaged over block, enveloped or long cycle averaged trailing RMS, pitch detect, and probably something like first deriv of the trailing RMS although that’s close to enveloped I guess) and then you can just route those modulations wherever you want.  That is a very interesting approach.

**Comment by mkruselj** [comment]
Hm I am not sure if taking up an oscillator would be absolutely required for this? You would just work on the audio input directly, taking it without any gate/envelope and process it into those modulation sources?

**Comment by baconpaul** [comment]
oh yeah like the vocoder does. of course.
